pyiceberg/catalog/__init__.py

```diff
@@ -512,6 +516,22 @@ def _check_for_overlap(removals: Optional[Set[str]], updates: Properties) -> Non
             if overlap:
                 raise ValueError(f"Updates and deletes have an overlap: {overlap}")
 
+    @staticmethod
+    def _convert_schema_if_needed(schema: Union[Schema, "pa.Schema"]) -> Schema:
+        try:
```

Comment:
nit: it might be easier to first type check `Schema`, then `pa.Schema`, else ValueError

Reply:
This approach is a result of an earlier discussion with @Fokko : https://github.com/apache/iceberg-python/pull/305#discussion_r1466705205

Reply:
Are you suggesting the following instead?
```
        if isinstance(schema, Schema):
            return schema
        try:
            import pyarrow as pa

            from pyiceberg.io.pyarrow import _ConvertToIcebergWithFreshIds, pre_order_visit_pyarrow

            if isinstance(schema, pa.Schema):
                schema: Schema = pre_order_visit_pyarrow(schema, _ConvertToIcebergWithFreshIds())  # type: ignore
                return schema
        except ModuleNotFoundError:
            pass
         raise ValueError(f"{type(schema)=} must be pyiceberg.schema.Schema")
```
Maybe it is better, since we avoid checking for the import if the type is already correct

Reply:
Yep, that's what I meant

Reply:
And also update ValueError message to include pyarrow.Schema

Reply:
Sounds good. These are fantastic nits, thank you! :)